symfit/core/fit.py

```diff
@@ -1372,31 +1353,52 @@ def _determine_minimizer(self):
             return BFGS
 
     @staticmethod
-    def _determine_objective(model, objective, minimizer):
+    def _determine_objective(model, objective, minimizer, bound_arguments):
         """
         Determine the most suitable objective on the basis of the problem at
-        hand.
+        hand. This could modify ``bound_arguments`` in place accordingly if
+        required!
 
         :param model: :class:`symfit.core.fit.BaseModel` under consideration.
         :param objective: objective provided to :class:`symfit.core.fit.Fit` by
             the user, or ``None``.
         :param minimizer: :class:`~symfit.core.minimizers.BaseMinimizer`
             provided by the user, or  ``None``
+        :param bound_arguments: Instance of :class:`inspect.BoundArguments`.
         :return: a subclass of `BaseObjective`.
         """
         if objective is not None:
+            # Client is king, return the clients objective. For LogLikelihood
+            # though, we do check if the data is compatible with likelihood
+            # fitting.
+            if objective is LogLikelihood or objective is MinimizeModel or \
+                    isinstance(objective, (MinimizeModel, LogLikelihood)):
+                # Set dependent vars and corresponding sigmas to None.
+                for var in model.dependent_vars + list(model.sigmas.values()):
+                    if var.name not in bound_arguments.arguments:
+                        bound_arguments.arguments[var.name] = None
+                    else:
+                        raise TypeError(
+                            'A value was provided for `{}`, however for {} '
+                            'fitting the dependent variable cannot have a value '
+                            'assigned to it.'.format(objective, var.name)
+                        )
             return objective
         else:
             if minimizer is MINPACK:
                 # MINPACK is considered a special snowflake, as its API has to
-                # be considered seperately and has its own non standard
+                # be considered separately and has its own non standard
                 # objective function.
                 return VectorLeastSquares
-            # Param only scalar Model -> the model is the objective.
-            elif len(model.independent_vars) == 0 and len(model) == 1:
-                # No data provided means a simple minimization of the Model
-                # parameters is requested, not a fit.
-                return MinimizeModel
+            if len(model) == 1:
+                if len(model.independent_vars) == 0 and model.dependent_vars[0].name not in bound_arguments.arguments:
```

Comment:
Combine the conditions?

Reply:
I decided to go for two steps otherwise this line would become way to long, even for my taste. But I will reordered them slightly to make it even more compact.